### PR TITLE
ci: add gov module to interchaintest folder

### PR DESCRIPTION
**Pull request by dasanchez**
## Description

Adds gov module tests to delegator folder:
- Proposal submission
- Proposal deposits
- Proposal vote
- Proposal weighted vote
- Gov module param change
- Messages from gov module account

Successful workflow run:
https://github.com/cosmos/gaia/actions/runs/11732771239

---

### Author Checklist

I have...

- [X] Included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [X] Targeted the correct branch (see [PR Targeting](https://github.com/cosmos/gaia/blob/main/CONTRIBUTING.md#pr-targeting))
- [ ] Provided a link to the relevant issue or specification
- [X] Reviewed "Files changed" and left comments if necessary 
- [x] Confirmed all CI checks have passed

### Reviewers Checklist

I have...

- [ ] Confirmed the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] Confirmed all author checklist items have been addressed
- [ ] Confirmed that this PR does not change production code

### CODE REVIEW

File: tests/interchain/delegator/gov_test.go [hunk 1]
```diff
@@ -0,0 +1,280 @@
...
+	actual_no_weight := vote.Get("options.#(option==\"VOTE_OPTION_NO\").weight")
+	// chainsuite.GetLogger(s.GetContext()).Sugar().Infof("%s", actual_no_weight.String())
+	s.Require().Equal(float64(noWeight), actual_no_weight.Float())
+	actual_abstain_weight := vote.Get("options.#(option==\"VOTE_OPTION_ABSTAIN\").weight")
+	// chainsuite.GetLogger(s.GetContext()).Sugar().Infof("%s", actual_abstain_weight.String())
+	s.Require().Equal(float64(abstainWeight), actual_abstain_weight.Float())
+	actual_veto_weight := vote.Get("options.#(option==\"VOTE_OPTION_NO_WITH_VETO\").weight")
+	// chainsuite.GetLogger(s.GetContext()).Sugar().Infof("%s", actual_veto_weight.String())
+	s.Require().Equal(float64(vetoWeight), actual_veto_weight.Float())
+}
```
Comment (fastfadingviolets) [inline tests/interchain/delegator/gov_test.go]: all these are the ones that use the proposal you submitted in Test01 right? I'd suggest gathering all of them together into one test func and then using s.Run() to split up the individual stories

**Review by fastfadingviolets — CHANGES_REQUESTED** [review]
Ideally we'd like to be able to run the test funcs independently of one another. So the order shouldn't matter that much. I suggest consolidating the ones that all use the same proposal into one function and then removing the number indicator from the other ones. This way there's no global state or anything, and the tests are independent.

File: tests/interchain/delegator/gov_test.go (hunk 1 above)
Comment (dasanchez) [reply tests/interchain/delegator/gov_test.go]: Thanks @fastfadingviolets ! I decoupled all tests.

**Comment by dasanchez** [comment]
Successful workflow run:
https://github.com/cosmos/gaia/actions/runs/11829964071

**Review by fastfadingviolets — APPROVED** [review]

**Review by MSalopek — APPROVED** [review]

**Pull request merged**